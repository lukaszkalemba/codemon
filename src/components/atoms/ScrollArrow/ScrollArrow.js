import React from 'react';
import useWindowSize from 'hooks/useWindowSize';
import Icon from 'components/atoms/Icon';
import arrow from 'assets/svgs/icon_scrolldown-arrow.svg';
import S from './ScrollArrow.styles';

const ScrollArrow = () => {
  const size = useWindowSize();

  const handleClick = () => {
    window.smoothScroll.scrollTo(0, size.height + 100, 1000);
  };

  return (
    <S.Button onClick={handleClick}>
      <S.Info>Przesuń w dół</S.Info>
      <Icon src={arrow} />
    </S.Button>
  );
};

export default ScrollArrow;
