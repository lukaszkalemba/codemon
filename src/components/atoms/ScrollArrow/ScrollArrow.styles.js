import styled from 'styled-components';

const Button = styled.button`
  display: none;
  border: none;
  background-color: inherit;
  width: 50px;
  margin-left: auto;
  margin-top: auto;
  cursor: pointer;
  position: relative;
  z-index: ${({ theme }) => theme.zindex.positive1};
  transition: transform 250ms ease-in-out;
  outline: none;

  :hover,
  :focus {
    transform: translateY(10px);
    svg {
      opacity: 1;
    }
  }

  ${({ theme }) => theme.media.laptop`
    width: 75px;
    height: 150px;
    display: block;
  `}
  svg {
    width: 100%;
    height: 100%;
    opacity: 0.6;
    transition: opacity 250ms ease-in-out;
  }
`;

const Info = styled.p`
  position: absolute;
  width: 120px;
  transform: rotate(90deg);
  color: ${({ theme }) => theme.color.white};
  right: -17px;
  top: -25px;
  opacity: 0.7;
`;

export default { Button, Info };
