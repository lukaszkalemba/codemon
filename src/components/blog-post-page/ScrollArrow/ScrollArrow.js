import React from 'react';
import useWindowSize from 'hooks/useWindowSize';
import Icon from 'components/common/Icon';
import arrow from 'assets/svgs/icon_scrolldown-arrow.svg';
import S from './ScrollArrow.styles';

const ScrollArrow = () => {
  const size = useWindowSize();

  const handleClick = () => {
    window.smoothScroll.scrollTo(0, size.height + size.height / 10, 1000);
  };

  return (
    <S.Wrapper onClick={handleClick}>
      <Icon src={arrow} />
    </S.Wrapper>
  );
};

export default ScrollArrow;
