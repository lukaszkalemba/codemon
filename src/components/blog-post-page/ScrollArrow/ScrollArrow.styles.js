import styled from 'styled-components';

const Wrapper = styled.div`
  width: 50px;
  height: 100px;
  margin-left: auto;
  margin-top: auto;

  ${({ theme }) => theme.media.laptop`
    width: 75px;
    height: 150px;
  `}

  svg {
    width: 100%;
    height: 100%;
  }
`;

export default { Wrapper };
