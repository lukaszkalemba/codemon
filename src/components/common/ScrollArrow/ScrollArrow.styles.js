import styled from 'styled-components';

const Wrapper = styled.button`
  display: block;
  border: none;
  background-color: inherit;
  width: 50px;
  height: 100px;
  margin-left: auto;
  margin-top: auto;
  cursor: pointer;
  position: relative;
  z-index: ${({ theme }) => theme.zindex.positive1};
  transition: transform 250ms ease-in-out, opacity 250ms ease-in-out;
  opacity: 0.75;
  outline: none;

  :hover {
    transform: translateY(10px);
    opacity: 1;
  }

  ${({ theme }) => theme.media.laptop`
    width: 75px;
    height: 150px;
  `}
  svg {
    width: 100%;
    height: 100%;
  }
`;

export default { Wrapper };
