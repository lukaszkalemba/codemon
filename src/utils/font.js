export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
