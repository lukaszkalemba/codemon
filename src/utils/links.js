export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
