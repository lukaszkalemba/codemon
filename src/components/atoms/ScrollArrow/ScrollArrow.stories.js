import React from 'react';
import ScrollArrow from './ScrollArrow';

const handleArrowClick = () => {};

export const scrollArrow = () => <ScrollArrow onClick={handleArrowClick} />;

export default {
  component: ScrollArrow,
  title: 'Atoms|ScrollArrow',
};
