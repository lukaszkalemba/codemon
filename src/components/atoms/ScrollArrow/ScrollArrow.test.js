import React from 'react';
import { render } from 'utils/test-utils';
import ScrollArrow from './ScrollArrow';

describe('<ScrollArrow />', () => {
  it('matches snapshot', () => {
    const { container } = render(<ScrollArrow />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
