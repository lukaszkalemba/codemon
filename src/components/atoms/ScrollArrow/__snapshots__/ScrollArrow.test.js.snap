// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollArrow /> renders correctly and matches snapshot 1`] = `
<button
  class="sc-AxirZ bPTMKP"
>
  <p
    class="sc-AxiKw fmBTKk"
  >
    Przesuń w dół
  </p>
  <div
    class="sc-AxjAm cQfdzG"
  >
    <svg
      fill="none"
      height="16"
      viewBox="0 0 65 16"
      width="65"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 1H61L42.5 14.5"
        stroke="white"
        stroke-width="2"
      />
    </svg>
  </div>
</button>
`;
